[dlatikay 20191221] support for newer wiimote motionplus

dlatikaynen opened the pull request:
support for newer wiimote motionplus, which reveal their name only after authentication.
Also, those wiimote MotionPlus have the same vendor product code as a Wii U Pro Controller (0x330) and must be distinguished earlier - otherwise, the icon on the panel changes from pro to wiimote only later when the report is parsed as the directinput emulation starts.

dlatikaynen commented:
VS screwed up the code indentation immediately before the first commit and I failed to check the diffs before committing, therefore I reverted it and did a second, clean commit.

KeyPuncher commented on Shared/Windows/WinBtStream.cs:
```diff
@@ -289,13 +292,71 @@ public static List<DeviceInfo> GetPaths()
+                            // note: newer Wiimotes may identify as ProControllers via their ProductID
+                            // they reveal to be Wiimote only in their extension/status report
+                            // so make one roundtrip to ask the controller for its actual type
+                            using (var _stream = new WinBtStream(newDeviceInfo.DevicePath))
```
There is already a measure in place that will make the icon be whatever the last controller type was detected for the same controller so they really just have to do a first time connect. If we're unsure about the controller type would could add a "?" badge on the icon to indicate it may or may not be the controller indicated.

dlatikaynen commented:
kept only the improvement where controllers reveal their name only after BT auth.
reverted the (diagnostic only) additions showing the internal device name and serial number on prop dlg
added the suggested improvement of not connecting up-front; idea: as the user clicks "identify" we use that roundtríp to get a hold of the final device type, too. goes well with the UX meaning of "identify".
